### DISCUSSION TITLE
Bug: exception_handlers are not applied when debug flag is False

### DISCUSSION THREAD

**Discussion by TheZwieback** [litestar-org/litestar#3581] [Q&A] [answered]
### Description

I want to configure a custom exception handler for the following use case. I want to run the app in production but log stack traces of internal server errors but other HTTPException should not be logged with stack trace as they are usually triggered specifically and converted to a proper response by Litestar.
In this way I can raise HTTPExceptions for issues inside my code without having to create explicit responses. Also I have the case that I want to trigger HTTPExceptions in a dependency injection directly in one place without having to duplicate error handling in each endpoint where it is used.
Anyways, the current issue is that I cannot get the exception handler to trigger when debug is set to False in the app.

### MCVE

```python
from litestar.exceptions import HTTPException
from litestar.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
from litestar import Request, Response

from litestar import Litestar, post
import logging
from litestar.exceptions import LitestarException


def exception_handler(_: Request, exc: Exception) -> Response:
    if isinstance(exc, HTTPException):
        status_code = getattr(exc, "status_code", HTTP_500_INTERNAL_SERVER_ERROR)
        detail = getattr(exc, "detail", "")

        return Response(
            content={"detail": detail},
            status_code=status_code,
        )

    logging.exception(exc)
    status_code = HTTP_500_INTERNAL_SERVER_ERROR
    detail = {"special_detail": "Internal Server Error"}

    return Response(content=detail, status_code=status_code)


@post(
    "/test",
)
async def test() -> Response[str]:
    1 / 0
    return Response("test")


app = Litestar(
    route_handlers=[test],
    exception_handlers={LitestarException: exception_handler},
)
```

### Steps to reproduce

```bash
1. Run the code
2. Trigger the endpoint
3. response body shows detail property instead of special_detail
```

### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```

### Logs

```bash
INFO:     Started server process [48233]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:38254 - "GET /schema/swagger HTTP/1.1" 200 OK
INFO:     127.0.0.1:38254 - "POST /test HTTP/1.1" 500 Internal Server Error
```

### Litestar Version

2.9.0

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3579">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3579/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3579/pledge.svg">
</picture>
</a>

**Comment by Alc-Alc** [comment]
Not sure I understood this, you are applying the exception handler to `LitestarException` but your route handler raises a different exception, so how is it you expect it to trigger?

Your MCVE never applies the handler even if debug = True for the same reason stated above

**Comment by TheZwieback** [comment, marked as answer]
Thanks, I see my mistake. I assumed the LitestarException map would catch all exceptions for some reason. I checked the code and if I use the correct error mapping it works. Here is the working example for completeness:
```py
from litestar.exceptions import HTTPException
from litestar.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
from litestar import Request, Response, Litestar, post

import logging


def exception_handler(_: Request, exc: Exception) -> Response:
    if isinstance(exc, HTTPException) and exc.status_code != HTTP_500_INTERNAL_SERVER_ERROR:
        status_code = getattr(exc, "status_code", HTTP_500_INTERNAL_SERVER_ERROR)
        detail = getattr(exc, "detail", "")

        return Response(
            content={"detail": detail},
            status_code=status_code,
        )

    logging.exception(exc)
    status_code = HTTP_500_INTERNAL_SERVER_ERROR
    detail = {"special_detail": "Internal Server Error"}

    return Response(content=detail, status_code=status_code)


@post(
    "/test",
)
async def test() -> Response[str]:
    1 / 0
    return Response("test")


app = Litestar(
    route_handlers=[test],
    exception_handlers={Exception: exception_handler},
)

```
Now I can run with stacktraces in production without debug mode leaking the traces
